ls1intum/Artemis
# [Bugfix] Fix duplicate testcase entries

### Checklist

### Motivation and Context


When a testcase was renamed from e.g. `testMergeSort` to `testmergeSort`, it would be treated as a new testcase and saved to the database. But the database handles both testcase names as case-insensitive, so this was causing an error.

Fixes: #2505 

### Description

When comparing testcases we compare them by using the lowerCase.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a programming exercise 
4. Go to Edit in Editor and change one of the test names (e.g. `testMergeSort`) to a lowercase version
5. The Programming Exercise Details should display the latest result correctly
6. The Submission View of the Solution Repo should also be consistent and show the latest result and has no empty row
7. Now create a different test class (e.g. MyTest.java) and duplicate the test `testMergeSort`. 
8. After submitting as an instructor or tutor you should receive a notification that duplicated tests were found. It is also shown in the result string and as an additional feedback element.

### Test Coverage



ProgrammingExerciseGradingService: 93%

### Screenshots


#### These errors should not occur: 
![image](https://user-images.githubusercontent.com/33484318/110937548-cf08dc80-8332-11eb-8c0c-18bee5165b3c.png)
#### This shows that a result was not attached to the submission:
![image](https://user-images.githubusercontent.com/33484318/110937600-e2b44300-8332-11eb-9488-c5cee2336473.png)

#### It should look like this:
![image](https://user-images.githubusercontent.com/33484318/110948211-5b220080-8341-11eb-961a-18964db3e183.png)

### Duplicated Test Cases
#### Result String of Template and Solution Repo:
![image](https://user-images.githubusercontent.com/33484318/111071947-c7833800-84d8-11eb-9db0-8d8fae6708fa.png)

#### Feedback element:
![image](https://user-images.githubusercontent.com/33484318/111071922-af131d80-84d8-11eb-8d70-6de8fb70a809.png)

#### Received notification as an instructor:
![image](https://user-images.githubusercontent.com/33484318/111071895-90ad2200-84d8-11eb-88cf-76446498d4d0.png)
